db/src/main/java/org/odk/collect/db/sqlite/MigrationListDatabaseMigrator.kt

```diff
@@ -0,0 +1,25 @@
+package org.odk.collect.db.sqlite
+
+import android.database.sqlite.SQLiteDatabase
+
+abstract class MigrationListDatabaseMigrator(private val databaseVersion: Int, vararg migrations: ((SQLiteDatabase) -> Unit)) :
```

Comment:
This hierarchy could be flattened once `MigrationListDatabaseMigrator` is used everywhere (both in the forms and instances databases). At that point, we could drop `DatabaseMigrator` and simply rename `MigrationListDatabaseMigrator` to `DatabaseMigrator`.
If that's the plan, we might want to add a comment to remind ourselves of it.

Reply:
I'm not sure to be honest: we use `DatabaseMigrator` in places like `DatabaseMigratorSQLiteOpenHelper` where we really just want the `onCreate`/`onUpgrade` methods. I'd still see `MigrationListDatabaseMigrator` as a "convenience" implementation really.